crates/agglayer-config/src/storage.rs

```diff
@@ -0,0 +1,153 @@
+use std::path::Path;
+use std::path::PathBuf;
+
+use serde::Deserialize;
+use serde::Serialize;
+
+const STORAGE_DIR: &str = "storage";
+const METADATA_DB_NAME: &str = "metadata";
+const PENDING_DB_NAME: &str = "pending";
+const STATE_DB_NAME: &str = "state";
+const EPOCHS_DB_PATH: &str = "epochs";
+
+/// Configuration for the storage.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct StorageConfig {
+    /// Custom metadata storage path or inferred from the db path.
+    pub metadata_db_path: PathBuf,
+    /// Custom pending storage path or inferred from the db path.
+    pub pending_db_path: PathBuf,
+    /// Custom state storage path or inferred from the db path.
+    pub state_db_path: PathBuf,
+    /// Custom epochs storage path or inferred from the db path.
+    pub epochs_db_path: PathBuf,
+}
```

Comment:
At risk of overcomplicating things... Could there be an intermediate `storage_dir` config, defaulting to `"storage"`?

Then, for example, the path to the epochs db would be:
```rust
base_dir.join(storage_dir).join(epochs_db_path)
```

This should work as expected even in presence of absolute paths as noted in the other comment.

It allows the user to change the location of all the databases without changing `base_dir` or individual database paths. Having the databases in an extra directory may be useful if the user wants to nuke the database without deleting other stuff that might live in the base directory (configs, logs, whatever).

Reply:
That's exactly what the `Deserialize` implementation do. Currently you just need to define the `db-path` and all the other path are inferred from it. 
I removed the fact that one can define a custom path for one database only, if we need it we'll add it later.

The configuration looks like this now:

```toml

[storage]
db-path = "./database"
```